packages/elements/src/components.d.ts

```diff
@@ -2194,6 +2195,7 @@ declare namespace LocalJSX {
         "filled"?: boolean;
         /**
           * The name of the icon of this element.
+          * @deprecated This property is deprecated and will be removed with the next major release. Instead, use the `icon-leading` slot.
```

Comment:
Do we really want to remove it? I'd say we keep both. The prop is very convenient for many use-cases and the slot provides flexibility for edge cases.